roles/kustomize_deploy/tasks/install_controlplane.yml

```diff
@@ -0,0 +1,97 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure that values.yaml file for control plane deployment exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_kustomize_deploy_nncp_values_src_file }}"
+  register: _cifmw_kustomize_deploy_nncp_values_exists
+  failed_when: not _cifmw_kustomize_deploy_cp_values_exists.stat.exists
+
+- name: Copy values.yaml file to NNCP directory
+  ansible.builtin.copy:
+    remote_src: true
+    src: "{{ cifmw_kustomize_deploy_nncp_values_src_file }}"
+    dest: "{{ cifmw_kustomize_deploy_nncp_values_dest_file }}"
+    mode: "0644"
+
+- name: Generate the NNCP kustomization file
+  ansible.builtin.copy:
+    content: >-
+      {{
+        lookup
+        (
+          'kubernetes.core.kustomize',
+          dir=cifmw_kustomize_deploy_nncp_source_files
+        )
+      }}
+    dest: "{{ cifmw_kustomize_deploy_nncp_dest_file }}"
+    mode: "0644"
+
+- name: Apply the NNCP CR
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: openstack
+    state: present
+    src: "{{ cifmw_kustomize_deploy_nncp_dest_file }}"
+
+- name: Wait for NNCP resources
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    kind: nncp
+    namespace: openstack
+    label_selectors:
+      - osp/nncm-config-type = standard
+    wait: true
+    wait_condition:
+      type: SuccessfullyConfigured
+      status: "True"
+    wait_timeout: 300
+
+- name: Generate the control plane kustomization file
+  ansible.builtin.copy:
+    content: >-
+      {{
+        lookup
+        (
+          'kubernetes.core.kustomize',
+          dir=cifmw_kustomize_deploy_cp_source_files
+        )
+      }}
+    dest: "{{ cifmw_kustomize_deploy_cp_dest_file }}"
+    mode: "0644"
+
+- name: Apply the control plane CR
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: openstack
+    state: present
+    src: "{{ cifmw_kustomize_deploy_cp_dest_file }}"
+
+- name: Wait for control plane resources
```

Comment:
This thing doesn't wait. It's poorly handling errors:
```
TASK [kustomize_deploy : Wait for control plane resources kubeconfig={{ cifmw_openshift_kubeconfig }}, kind=osctlplane, name=controlplane, namespace=openstack, wait=True, wait_condition={'type': 'Ready', 'status': 'True'}, wait_timeout=600] ***
Friday 02 February 2024  13:32:40 -0500 (0:00:01.300)       0:00:31.196 ******* 
ok: [localhost]

TASK [kustomize_deploy : Debug var=_output] **************************************************************************************************************************************************
Friday 02 February 2024  13:32:43 -0500 (0:00:03.508)       0:00:34.704 ******* 
ok: [localhost] => {
    "_output": {
        "api_found": false,
        "changed": false,
        "failed": false,
        "msg": "Failed to find API for resource with apiVersion \"v1\" and kind \"osctlplane\"",
        "resources": []
    }
}
```
The issue, the apiVersion, this thing, not like oc, is not able to properly consume api-resource info from the cluster and just defaults to a dumb v1 string.
This may do the job if we register the output of the CP apply:
```

- name: Wait for control plane resources
  vars:
    _applied_osctlplane_cr: >-
      {{
        (
          _cifmw_kustomize_deploy_cp_apply_out.result.results |
          selectattr('result.kind', 'equalto', 'OpenStackControlPlane') |
          first
        ).result
      }}
  kubernetes.core.k8s_info:
    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
    api_version: "{{ _applied_osctlplane_cr.apiVersion }}"
    kind: "{{ _applied_osctlplane_cr.kind }}"
    name: "{{ _applied_osctlplane_cr.metadata.name }}"
    namespace: "{{ _applied_osctlplane_cr.metadata.namespace}}"
    wait: true
    wait_condition:
      type: Ready
      status: "True"
    wait_timeout: 600
  register: _output

```

But... it's still not waiting, even if the params are ok:
```
ok: [localhost] => {
    "api_found": true,
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": null,
            "api_version": "core.openstack.org/v1beta1",
            "ca_cert": null,
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "hidden_fields": null,
            "host": null,
            "impersonate_groups": null,
            "impersonate_user": null,
            "kind": "OpenStackControlPlane",
            "kubeconfig": "/home/zuul/.kube/config",
            "label_selectors": [],
            "name": "controlplane",
            "namespace": "openstack",
            "no_proxy": null,
            "password": null,
            "persist_config": null,
            "proxy": null,
            "proxy_headers": null,
            "username": null,
            "validate_certs": null,
            "wait": true,
            "wait_condition": {
                "reason": null,
                "status": true,
                "type": "Ready"
            },
            "wait_sleep": 5,
            "wait_timeout": 600
        }
    },

```
IMHO: Ditch the entire usage of the module. Zero issues calling oc wait for this use case.

Reply:
Something like this seems to work (tested in my box):
```
- name: Wait for OpenStack controlplane to be deployed
  environment:
    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
    PATH: "{{ cifmw_path }}"
  vars:
    _applied_osctlplane_cr: >-
      {{
        (
          _cifmw_kustomize_deploy_cp_apply_out.result.results |
          selectattr('result.kind', 'equalto', 'OpenStackControlPlane') |
          first
        ).result
      }}
  ansible.builtin.command:
    cmd: >-
      oc wait {{ _applied_osctlplane_cr.kind }}
      {{ _applied_osctlplane_cr.metadata.name }}
      --namespace={{ _applied_osctlplane_cr.metadata.namespace }}
      --for=condition=ready
      --timeout=600s

```

PS: The previous apply seems to violate idempotency, so no need to mess with the `changed_when` condition of this command.